version: '3.8'
services:
  bot:
    build: .
    container_name: bot-app
    volumes:
      - ./src:/app
      - ./registry-fec-bot.xlsx:/app/registry-fec-bot.xlsx
    depends_on:
      db:
        condition: service_healthy
    environment:
      DATABASE_URL: postgresql://${DB_USER}:${DB_PASSWORD}@db:5432/${DB_NAME}
      BOT_TOKEN: ${BOT_TOKEN}
    command: ["python3", "bot.py"]
    networks:
      - bot-network
    restart: unless-stopped  
  db:
    image: postgres:latest
    container_name: db-bot
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USER}
      POSTGRES_DB: ${DB_NAME}
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 10s
      retries: 3
      start_period: 30s
      timeout: 10s  
    volumes:
      - ./database:/var/lib/postgresql/data
    networks:
      - bot-network
    restart: unless-stopped   

networks:
  bot-network:
    driver: bridge      